CONTRIBUTING.md

```diff
@@ -11,30 +11,98 @@ best way to contribute to the project, read on. This document will cover
+issues of each type.
 
-* Make sure you test against the latest released version. It is possible
-  we already fixed the bug you're experiencing.
+#### Bug Reports
 
-* Provide steps to reproduce the issue, along with your `.tf` files,
-  with secrets removed, so we can try to reproduce it. Without this,
-  it makes it much harder to fix the issue.
+ - [ ] __Test against latest release__: Make sure you test against the latest
+   released version. It is possible we already fixed the bug you're experiencing.
 
-* If you experienced a panic, please create a [gist](https://gist.github.com)
-  of the *entire* generated crash log for us to look at. Double check
-  no sensitive items were in the log.
+ - [ ] __Search for possible duplicate reports__: It's helpful to keep bug
+   reports consolidated to one thread, so do a quick search on existing bug
+   reports to check if anybody else has reported the same thing. You can scope
+   searches by the label "bug" to help narrow things down.
 
-* Respond as promptly as possible to any questions made by the Terraform
-  team to your issue. Stale issues will be closed.
+ - [ ] __Include steps to reproduce__: Provide steps to reproduce the issue,
+   along with your `.tf` files, with secrets removed, so we can try to
+   reproduce it. Without this, it makes it much harder to fix the issue.
+
+ - [ ] __For panics, include `crash.log`__: If you experienced a panic, please
+   create a [gist](https://gist.github.com) of the *entire* generated crash log
+   for us to look at. Double check no sensitive items were in the log.
+
+#### Feature Requests
+
+ - [ ] __Search for possible duplicate requests__: It's helpful to keep requests
+   consolidated to one thread, so do a quick search on existing requests to
+   check if anybody else has reported the same thing. You can scope searches by
+   the label "enhancement" to help narrow things down.
+
+ - [ ] __Include a use case description__: In addition to describing the
+   behavior of the feature you'd like to see added, it's helpful to also lay
+   out the reason why the feature would be important and how it would benefit
+   Terraform users.
+
+#### Questions
+
+ - [ ] __Search for answers in Terraform documentation__: We're happy to answer
+   questions in GitHub Issues, but it helps reduce issue churn and maintainer
+   workload if you work to find answers to common questions in the
+   documentation. Often times Question issues result in documentation updates
+   to help future users, so if you don't find an answer, you can give us
+   pointers for where you'd expect to see it in the docs.
 
 ### Issue Lifecycle
 
 1. The issue is reported.
 
 2. The issue is verified and categorized by a Terraform collaborator.
-   Categorization is done via tags. For example, bugs are marked as "bugs".
+   Categorization is done via GitHub labels. We generally use a two-label
+   system of (1) issue/PR type, and (2) section of the codebase. Type is
```

Comment:
I know this may sound like a nitpick, but often I'm questioning myself whether it's enough to tag issue/PR with `docs` & type (provider/provisioner/core) or whether I'm also expected to tag this with `enhancement` or `bug` depending on the nature of docs update - i.e. undocumented or incorrectly documented feature would be a bug, adding a new section would be `enhancement`.